From b804a741d5cdb5d1b761de9ad053ba916fa08636 Mon Sep 17 00:00:00 2001
From: Paul Eggert <eggert@cs.ucla.edu>
Date: Tue, 22 Dec 2020 15:11:11 -0800
Subject: [PATCH] Improve doc of zic coalescing zone+DST transitions

* zic.8: Improve documentation for how certain transitions
are coalesced.  Also, use current data for America/Menominee
instead an America/Grand_Turk entry that not in current data.
Improvements suggested by Tim Parenti.
---
 zic.8 | 36 +++++++++++++++++++++---------------
 1 file changed, 21 insertions(+), 15 deletions(-)

diff --git a/zic.8 b/zic.8
index 2273dad..60262d7 100644
--- a/zic.8
+++ b/zic.8
@@ -590,35 +590,41 @@ first transition into standard time.
 In a single zone it is an error if two rules take effect at the same
 instant, or if two zone changes take effect at the same instant.
 .PP
-If a continuation line subtracts from the UT offset after a transition that
-would be interpreted to be later if using the continuation line's rules, the
+If a continuation line subtracts
+.I N
+seconds from the UT offset after a transition that would be
+interpreted to be later if using the continuation line's UT offset and
+rules, the
 .q "until"
-time of the earlier zone or continuation line is interpreted according
-to the continuation line's rules.
+time of the previous zone or continuation line is interpreted
+according to the continuation line's UT offset and rules, and any rule
+that would otherwise take effect in the next
+.I N
+seconds is instead assumed to take effect simultaneously.
 For example:
 .br
 .ne 7
 .nf
 .in +2m
-.ta \w'# Rule\0\0'u +\w'NAME\0\0'u +\w'FROM\0\0'u +\w'max\0\0'u +\w'\*-\0\0'u +\w'Mar\0\0'u +\w'Sun>=8\0\0'u +\w'2:00\0\0'u +\w'SAVE\0\0'u
+.ta \w'# Rule\0\0'u +\w'NAME\0\0'u +\w'FROM\0\0'u +\w'2006\0\0'u +\w'\*-\0\0'u +\w'Oct\0\0'u +\w'lastSun\0\0'u +\w'2:00\0\0'u +\w'SAVE\0\0'u
 .sp
 # Rule	NAME	FROM	TO	\*-	IN	ON	AT	SAVE	LETTER/S
-Rule	US	2007	max	\*-	Mar	Sun>=8	2:00	1:00	D
-Rule	US	2007	max	\*-	Nov	Sun>=1	2:00	0	S
-.ta \w'Zone\0\0America/Grand_Turk\0\0'u +\w'STDOFF\0\0'u +\w'RULES\0\0'u +\w'FORMAT\0\0'u
+Rule	US	1967	2006	-	Oct	lastSun	2:00	0	S
+Rule	US	1967	1973	-	Apr	lastSun	2:00	1:00	D
+.ta \w'Zone\0\0America/Menominee\0\0'u +\w'STDOFF\0\0'u +\w'RULES\0\0'u +\w'FORMAT\0\0'u
 # Zone\0\0NAME	STDOFF	RULES	FORMAT	[UNTIL]
-Zone\0\0America/Grand_Turk	\*-4:00	\*-	AST	2018 Mar 11 2:00
-	\*-5:00	US	E%sT
+Zone\0\0America/Menominee	\*-5:00	\*-	EST	1973 Apr 29 2:00
+	\*-6:00	US	C%sT
 .sp
 .in
 .fi
-Here, an incorrect reading would be there were two clock changes on 2018-03-11,
-the first from 02:00 AST (\*-04) to 01:00 EST (\*-05),
-and the second an hour later from 02:00 EST (\*-05) to 03:00 EDT (\*-04).
+Here, an incorrect reading would be there were two clock changes on 1973-04-29,
+the first from 02:00 EST (\*-05) to 01:00 CST (\*-06),
+and the second an hour later from 02:00 CST (\*-06) to 03:00 CDT (\*-05).
 However,
 .B zic
-interprets this more sensibly as a single transition from 02:00 AST (\*-04) to
-02:00 EDT (\*-04).
+interprets this more sensibly as a single transition from 02:00 CST (\*-05) to
+02:00 CDT (\*-05).
 .PP
 A link line has the form
 .sp
-- 
1.8.3.1

